src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -833,6 +836,45 @@ class spell_warr_victory_rush : public SpellScript
     }
 };
 
+// Warrior: Thunder Clap (SpellID: 6343)
+class spell_warr_thunder_clap : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_WARRIOR_THUNDER_CLAP_SLOW,
+            SPELL_WARRIOR_THUNDER_CLAP_REND,
+            SPELL_WARRIOR_REND
+        });
+    }
+
+    void HandleSlow(SpellEffIndex /*effIndex*/)
+    {
+        // Apply movement speed reduction
+        if (Unit* target = GetHitUnit())
```

Comment:
there's no need to check for gethitunit in OnEffectHitTarget, it's never null at that point